packages/workflow/src/components/workflow-menu.js

```diff
@@ -0,0 +1,302 @@
+/**
+ * External dependencies
+ */
+import { Command, useCommandState } from 'cmdk';
+import clsx from 'clsx';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState, useEffect, useRef, useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import {
+	Modal,
+	TextHighlight,
+	__experimentalHStack as HStack,
+} from '@wordpress/components';
+import {
+	store as keyboardShortcutsStore,
+	useShortcut,
+} from '@wordpress/keyboard-shortcuts';
+import { Icon, search as inputIcon } from '@wordpress/icons';
+import { executeAbility, store as abilitiesStore } from '@wordpress/abilities';
+
+const inputLabel = __( 'Run abilities and workflows' );
+
+function WorkflowInput( { isOpen, search, setSearch, abilities } ) {
+	const workflowMenuInput = useRef();
+	const _value = useCommandState( ( state ) => state.value );
+	const selectedItemId = useMemo( () => {
+		// Find the ability whose label matches the selected value
+		const ability = abilities.find( ( a ) => a.label === _value );
+		return ability?.name;
+	}, [ _value, abilities ] );
+	useEffect( () => {
+		// Focus the workflow palette input when mounting the modal.
+		if ( isOpen ) {
+			workflowMenuInput.current.focus();
+		}
+	}, [ isOpen ] );
+	return (
+		<Command.Input
+			ref={ workflowMenuInput }
+			value={ search }
+			onValueChange={ setSearch }
+			placeholder={ inputLabel }
+			aria-activedescendant={ selectedItemId }
+		/>
+	);
+}
+
+/**
+ * @ignore
+ */
+export function WorkflowMenu() {
+	const { registerShortcut } = useDispatch( keyboardShortcutsStore );
+	const [ search, setSearch ] = useState( '' );
+	const [ isOpen, setIsOpen ] = useState( false );
+	const [ abilityOutput, setAbilityOutput ] = useState( null );
+	const [ isExecuting, setIsExecuting ] = useState( false );
+	const containerRef = useRef();
+
+	const abilities = useSelect( ( select ) => {
+		const allAbilities = select( abilitiesStore ).getAbilities();
+		return allAbilities || [];
```

Comment:
Does it make sense that this empty array is always the same one? I think it might be reasonable since we also use `abilities` later to memoize filtered abilities.